scenes/game/local_game.gd

```diff
@@ -2568,7 +2705,54 @@ func do_remaining_overdrive(performing_player : Player):
 		events += [create_event(Enums.EventType.EventType_AdvanceTurn, active_turn_player, 0)]
 
 	return events
+	
+func do_remaining_character_action(performing_player : Player):
+	var events = []
+	change_game_state(Enums.GameState.GameState_Boost_Processing)
+	while remaining_character_action_effects.size() > 0:
+		var effect = remaining_character_action_effects[0]
+		remaining_character_action_effects.erase(effect)
+		events += do_effect_if_condition_met(performing_player, -1, effect, null)
+		if game_state == Enums.GameState.GameState_PlayerDecision:
+			# Player has a decision to make, so stop mid-effect resolve.
+			break
+
+	if game_state != Enums.GameState.GameState_PlayerDecision:
```

Comment:
This probably will not work if we added something later with a character action that does something and then does a strike.
That puts the game in GameState_WaitForStrike.
Another case could be if a character action gives a bonus action, maybe something like Hilda's flip? That doesn't want to pass the turn.

I think for now just add in the strike condition here since that's such an obvious case, and we'll call it good. If we add more characters that need this we'll just update it then.

Reply:
Oh, maybe the bonus actions part will already handled by the bonus actions code.

Reply:
Yeah bonus actions should be fine; I'll also probably still keep the piece of setting active_character_action to false if a strike is happening

Reply:
Yes, definitely want to unset this flag to not mess things up later and we can pretty much assume strike will always mean the character action is over.